### PR TITLE
Update torch distribution wrappers for Bernoulli and Categorical

### PR DESCRIPTION
**Pull request by neerajprad**
Updates the Bernoulli and Categorical wrappers after [pytorch/pytorch#4448](https://github.com/pytorch/pytorch/pull/4448).

 - Updates the wrappers for categorical and bernoulli to use logits directly.
 - Updates `one_hot_categorical` to directly use the corresponding torch distribution instead of wrapping over `categorical`.

Testing - ran unit tests with `make test-torch-dist` locally.

### CODE REVIEW

File: pyro/distributions/torch/bernoulli.py [hunk 1]
```diff
@@ -4,25 +4,24 @@
...
+        x_shape = ps.size() if ps is not None else logits.size()
         event_dim = 1
         super(Bernoulli, self).__init__(torch_dist, x_shape, event_dim, *args, **kwargs)
 
     def sample(self):
         return super(Bernoulli, self).sample().type_as(self.torch_dist.probs)
 
     def batch_log_pdf(self, x):
-        return super(Bernoulli, self).batch_log_pdf(x.long())
+        return super(Bernoulli, self).batch_log_pdf(x)
```
Comment (fritzo) [inline pyro/distributions/torch/bernoulli.py]: I think you can simply delete this method now.

File: pyro/distributions/torch/bernoulli.py [hunk 2]
```diff
@@ -4,25 +4,24 @@
...
 
     def enumerate_support(self):
```
Comment (fritzo) [inline pyro/distributions/torch/bernoulli.py]: We should fix PyTorch to not convert this to `.long()` ([code](https://github.com/pytorch/pytorch/blob/master/torch/distributions/bernoulli.py#L67)) and then delete this method.

File: pyro/distributions/torch/one_hot_categorical.py [hunk 3]
```diff
@@ -1,44 +1,33 @@
...
 
     def sample(self):
-        ps = self.torch_dist.probs.data
-        zero = ps.new(self._sample_shape + ps.shape).zero_()
-        indices = super(OneHotCategorical, self).sample().data
-        if indices.dim() < zero.dim():
-            indices = indices.unsqueeze(-1)
-        one_hot = zero.scatter_(-1, indices, 1)
-        return Variable(one_hot)
+        return self.torch_dist.sample(self._sample_shape)
```
Comment (fritzo) [inline pyro/distributions/torch/one_hot_categorical.py]: Delete this method.

File: pyro/distributions/torch/one_hot_categorical.py [hunk 4]
```diff
@@ -1,44 +1,33 @@
...
 
     def batch_log_pdf(self, x):
```
Comment (fritzo) [inline pyro/distributions/torch/one_hot_categorical.py]: Delete this method?

File: pyro/distributions/torch/bernoulli.py [hunk 5]
```diff
@@ -4,25 +4,24 @@
...
     def __init__(self, ps=None, logits=None, *args, **kwargs):
-        ps = get_clamped_probs(ps, logits, is_multidimensional=False)
-        torch_dist = torch.distributions.Bernoulli(ps)
-        x_shape = ps.size()
+        torch_dist = torch.distributions.Bernoulli(probs=ps, logits=logits)
+        x_shape = ps.size() if ps is not None else logits.size()
         event_dim = 1
         super(Bernoulli, self).__init__(torch_dist, x_shape, event_dim, *args, **kwargs)
 
     def sample(self):
```
Comment (fritzo) [inline pyro/distributions/torch/bernoulli.py]: I think you can delete this method now.

File: pyro/distributions/torch/bernoulli.py (hunk 1 above)
Comment (neerajprad) [reply pyro/distributions/torch/bernoulli.py]: Didn't realize, will fix.

File: pyro/distributions/torch/one_hot_categorical.py (hunk 4 above)
Comment (neerajprad) [reply pyro/distributions/torch/one_hot_categorical.py]: We are doing the masking and appending the (1,) dim so not sure if we can delete this.

File: pyro/distributions/torch/bernoulli.py (hunk 2 above)
Comment (neerajprad) [reply pyro/distributions/torch/bernoulli.py]: Makes sense, will do.

**Comment by fritzo** [comment]
Actually the changes aren't exercised on travis, right? So I'm safe to merge before tests pass?

**Comment by neerajprad** [comment]
> Actually the changes aren't exercised on travis, right? So I'm safe to merge before tests pass?

That's true. The changes shouldn't affect the tests that run on travis.
